Tests/GitHubMilestones.tests.ps1

```diff
@@ -0,0 +1,178 @@
+        $cred = New-Object System.Management.Automation.PSCredential "<username is ignored>", $secureString
+        Set-GitHubAuthentication -Credential $cred
+
+        $script:ownerName = $env:avOwnerName
+        $script:organizationName = $env:avOrganizationName
+
+        $message = @(
+            'This run is executed in the AppVeyor environment.',
+            'The GitHub Api Token won''t be decrypted in PR runs causing some tests to fail.',
+            '403 errors possible due to GitHub hourly limit for unauthenticated queries.',
+            'Use Set-GitHubAuthentication manually. modify the values in Tests\Config\Settings.ps1,',
+            'and run tests on your machine first.')
+        Write-Warning -Message ($message -join [Environment]::NewLine)
+    }
+}
+
+Initialize-AppVeyor
+
+$script:accessTokenConfigured = Test-GitHubAuthenticationConfigured
+if (-not $script:accessTokenConfigured)
+{
+    $message = @(
+        'GitHub API Token not defined, some of the tests will be skipped.',
+        '403 errors possible due to GitHub hourly limit for unauthenticated queries.')
+    Write-Warning -Message ($message -join [Environment]::NewLine)
+}
+
+# Backup the user's configuration before we begin, and ensure we're at a pure state before running
+# the tests.  We'll restore it at the end.
+$configFile = New-TemporaryFile
+try
+{
+    Backup-GitHubConfiguration -Path $configFile
+    Reset-GitHubConfiguration
+    Set-GitHubConfiguration -DisableTelemetry # We don't want UT's to impact telemetry
+
+    # Define Script-scoped, readonly, hidden variables.
+
+    @{
+        defaultIssueTitle = "This is a test issue."
+        defaultMilestoneTitle1 = "This is a test milestone title #1."
+        defaultMilestoneTitle2 = "This is a test milestone title #2."
+        defaultEditedMilestoneTitle = "This is an edited milestone title."
+        defaultMilestoneDescription = "This is a test milestone description."
+        defaultEditedMilestoneDescription = "This is an edited milestone description."
+        defaultMilestoneDueOn = (Get-Date).AddYears(1).Date
+    }.GetEnumerator() | ForEach-Object {
+        Set-Variable -Force -Scope Script -Option ReadOnly -Visibility Private -Name $_.Key -Value $_.Value
+    }
+
+    Describe 'Creating, modifying and deleting milestones' {
+        $repo = New-GitHubRepository -RepositoryName ([Guid]::NewGuid().Guid) -AutoInit
+        $issue = New-GitHubIssue -Uri $repo.svn_url -Title $defaultIssueTitle
+
+        Context 'For creating a new milestone' {
+            $newMilestone = New-GitHubMilestone -Uri $repo.svn_url -Title $defaultMilestoneTitle1 -State "Closed" -DueOn $defaultMilestoneDueOn
+            $existingMilestone = Get-GitHubMilestone -Uri $repo.svn_url -Milestone $newMilestone.number
+
+            It "Should have the expected title text" {
+                $existingMilestone.title | Should be $defaultMilestoneTitle1
+            }
+
+            It "Should have the expected state" {
```

Comment:
My nitpick here is that these test names are generic and so don't tell you anything about what is actually being tested.  When you run the tests, you should see: 
```
For creating a new milestone:
    ...
    'Should be in the closed state'
```